### PR TITLE
MM-38011 - select enterprise for first subscribe

### PR DESCRIPTION
**Pull request by pablovelezvidal**
#### Summary
Keep enterprise option preselected on first subscribe via de purchase modal

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-38011

#### Screenshots
Captura de pantalla 2021-08-18 a las 23 14 30

#### Release Note
```release-note
NONE
```

### CODE REVIEW

**Comment by pablovelezvidal** [comment]
Reverting the changes added in https://github.com/mattermost/mattermost-webapp/pull/8419/files#diff-78a8a19706dbd2a4425dd72bdab0502ed7a2cef16365ab7030a5a0588927bf47R218 that causes that the build uses the June 9 server release

**Review by stevemudie — APPROVED** [review]
Looks good @pablovelezvidal !

Current plan was Enterprise and that's what was selected when I went to make the purchase as expected.

File: .circleci/config.yml [hunk 1]
```diff
@@ -215,7 +215,7 @@ jobs:
           at: ~/mattermost/
       - run:
           command: |
-            curl -f -o server.tar.gz https://pr-builds.mattermost.com/mattermost-server/$(echo "${CIRCLE_BRANCH}" | sed 's/pull\//PR-/g')/mattermost-enterprise-linux-amd64.tar.gz || curl -f -o server.tar.gz https://s3.amazonaws.com/releases.mattermost.com/gitlab/bundle/release-6.0/mattermost-enterprise-linux-amd64.tar.gz
+            curl -f -o server.tar.gz https://pr-builds.mattermost.com/mattermost-server/$(echo "${CIRCLE_BRANCH}" | sed 's/pull\//PR-/g')/mattermost-enterprise-linux-amd64.tar.gz || curl -f -o server.tar.gz https://s3.amazonaws.com/releases.mattermost.com/gitlab/bundle/master/mattermost-enterprise-linux-amd64.tar.gz
```
Comment (AGMETEOR) [inline .circleci/config.yml:218]: @pablovelezvidal is this intended?

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -11,7 +11,7 @@ variables:
 
 include:
   - project: mattermost/ci/mattermost-webapp
-    ref: release-6.0
+    ref: master
```
Comment (AGMETEOR) [inline .gitlab-ci.yml:14]: intended?

Comment (pablovelezvidal) [reply .gitlab-ci.yml:14]: yes, both intended reverting this unintended change https://github.com/mattermost/mattermost-webapp/pull/8642#issuecomment-901799779

File: components/purchase_modal/purchase_modal.tsx [hunk 3]
```diff
@@ -115,10 +115,13 @@ function findProductInDictionary(products: Dictionary<Product> | undefined, prod
...
 
-function getSelectedProduct(products: Dictionary<Product> | undefined, productId?: string | null) {
+function getSelectedProduct(products: Dictionary<Product> | undefined, productId?: string | null, isFreeTrial?: boolean | null) {
     const currentProduct = findProductInDictionary(products, productId);
+    if (isFreeTrial) {
+        return currentProduct;
+    }
     let nextSku = CloudProducts.PROFESSIONAL;
-    if (currentProduct?.sku === CloudProducts.PROFESSIONAL) {
+    if (currentProduct?.sku !== CloudProducts.STARTER) {
```
Comment (marianunez) [inline components/purchase_modal/purchase_modal.tsx:124]: Given the condition above for free trial, do we want to leave this condition as before? Not sure what the logic is here

Comment (pablovelezvidal) [reply components/purchase_modal/purchase_modal.tsx:124]: yes, you are right, with the isFreeTrial condition, the previous validation makes more sense. Updating it now.

**Review by marianunez — APPROVED** [review]

File: .gitlab-ci.yml (hunk 2 above)
Comment (AGMETEOR) [reply .gitlab-ci.yml:14]: Thanks for the context

**Review by AGMETEOR — APPROVED** [review]

File: .circleci/config.yml (hunk 1 above)
Comment (pablovelezvidal) [reply .circleci/config.yml:218]: yes, intended reverting this unintended change #8642 (comment)

**Comment by amyblais** [comment]
@pablovelezvidal Is this for v6.0?

**Comment by pablovelezvidal** [comment]
> @pablovelezvidal Is this for v6.0?

yes, it is!
@marianunez ^^

**Comment by marianunez** [comment]
> yes, it is!
> @marianunez ^^

This is actually also slated for Sept. 8th Cloud Release - this should not affect on-prem v6.0, there should be no need to cherry pick to cloud

**Pull request merged**